src/challonge.service.js

```diff
@@ -53,6 +49,30 @@ const fetchMembers = ({ api, organization }) => async function () {
     )(tournamentsDetails);
 };
 
+const fetchOpenTournamentsForMember = ({ api, organization }) => async function (memberEmailHash, { signedUp = undefined, includeUnderway = true } = {}) {
+    const tournaments = await fetchOpenTournaments({ api, organization })(includeUnderway);
+
+    const tournamentsDetailsPromises = R.pipe(
+        R.map(({ id }) => id),
+        R.map(fetchTournament({ api })),
+    )(tournaments);
+    const tournamentsDetails = await Promise.all(tournamentsDetailsPromises);
+
+    const filterFunc = R.cond([
+        [R.equals(true), () => R.equals(true)],
+        [R.equals(false), () => R.equals(false)],
+        [R.T, () => R.T],
+    ])(signedUp);
+
+    return R.pipe(
+        R.map(t => ({
```

Comment:
could use something like: `const foo = o => R.assoc('bar', 'foo' + o.foo, o);` instead of map spread

Reply:
https://stackoverflow.com/questions/34660745/ramda-js-setting-property-on-an-object-using-a-value-from-the-same-object

Reply:
True, but I think I prefer to not have prop names in strings, even though it's more verbose. (We had the same discussion regarding lenses, I think.)

Reply:
True, we did. Wait do we even need to map the property can we just roll the R.any into the filter?

Reply:
Or do we used for something other than the filter?

Reply:
That's how if was in my first iteration, but I split it like this because the the `tournaments` command is conditionally rendering based on `is_signed_up`.